pkg/controller/installation/installer.go

```diff
@@ -162,23 +162,61 @@ func (i *Installer) patchService(
 	labelsToInject map[string]string,
 	ownerReference *metav1.OwnerReference,
 ) runtime.Object {
-	appName, ok := labelsToInject[shipperv1.AppLabel]
-	if !ok {
-		// If we reach this point, it is fine to panic() since there's nothing
-		// much to do if we don't have the application label.
-		panic(fmt.Sprintf("Programmer error, label %q should always be present", shipperv1.AppLabel))
+
+	requiredLabels := []string{shipperv1.AppLabel}
+	for _, label := range requiredLabels {
+		if _, ok := labelsToInject[label]; !ok {
+			// If we reach this point, it is fine to panic() since there's nothing
+			// much to do if we don't have the mandatory labels.
+			panic(fmt.Sprintf("Programmer error, label %q should always be present", label))
```

Comment:
I think what you wrote elsewhere (returning an error in the case of a missing label) makes more sense than `panic()`-ing, even if this is what the code was doing before.